management/src/main/java/io/micronaut/management/endpoint/health/HealthEndpoint.java

```diff
@@ -79,8 +90,39 @@ public HealthEndpoint(HealthAggregator<HealthResult> healthAggregator,
     @Read
     public Single<HealthResult> getHealth(@Nullable Principal principal) {
         HealthLevelOfDetail detail = levelOfDetail(principal);
+        // due to backward comaptibility all HealthIndicators that are not qualified with Readiness
+        // e.g. are Liveness or without qualifier are returned by this resource
         return Single.fromPublisher(
-                healthAggregator.aggregate(healthIndicators, detail)
+                healthAggregator.aggregate(
+                        Arrays.stream(healthIndicators).
+                                filter(healthIndicator ->
+                                        !Arrays.asList(readinessHealthIndicators).contains(healthIndicator))
```

Comment:
I don't like this line but I expect that Array.asList will be somehow optimised by JIT

Reply:
You should probably at least assign the list to a local variable outside the scope. I would also move the filter to the constructor of the class as there is no need to do this calculation for each request

Reply:
yeah, that's right